### DISCUSSION TITLE
Is it normal that it's not possible to make a rowid on a subquery?

### DISCUSSION THREAD

**Discussion by florentfougeres** [duckdb/duckdb#14709] [Q&A]
I'd like to be able to use the `rowid` pseudo column on a subquery.

For example : 

```sql
D select rowid, * from building ;
```
| rowid | osm_type | building | geom                                                                |
|-------|----------|----------|---------------------------------------------------------------------|
| 0     | way      | yes      | \x02\x04\x00\x00\x00\x00\x00\x00\x91-A\xBF\x92S@Bf\x15A\x…         |
| 1     | way      | yes      | \x02\x04\x00\x00\x00\x00\x00\x00\x84DA\xBF~T@B\xB5\xF8@\x…         |
| 2     | way      | yes      | \x02\x04\x00\x00\x00\x00\x00\x00\xB3mA\xBF\x93S@B\xEB)A\x…         |
| 3     | way      | yes      | \x02\x04\x00\x00\x00\x00\x00\x00;\x7FA\xBF\xF5S@B+eA\xBF9…         |

```sql
D select rowid, * from (select * from building) ;
Binder Error: Referenced column "rowid" not found in FROM clause!
Candidate bindings: "unnamed_subquery.osm_id", "unnamed_subquery.full_id", "unnamed_subquery.building"
LINE 1: select rowid, * from (select * from building) ...
```

**Comment by Tishj** [comment]
I dont know how sqlite/postgres behave here, but in DuckDB pseudo-columns originate from the table, so this is expected

I'd be curious to know if other systems behave differently on this front, we might revisit this if that's the case

**Reply by florentfougeres** [reply]
The equivalent of `rowid` on postgres is ctid and the behavior is the same as the one I describe on DuckDB. 

On DuckDB as on postgres you have to use `row_number()` for a subquery, but this degrades performance too much, rowid is much faster, which is why I wanted to use `rowid`.
